contracts/funding/FundingFactory.sol

```diff
@@ -1,29 +1,92 @@
 
-pragma solidity ^0.5.0;
+pragma solidity ^0.6.0;
 
-import "@openzeppelin/upgrades/contracts/Initializable.sol";
+//import "@openzeppelin/upgrades/contracts/Initializable.sol";
 import "./Funding.sol";
-import "./chainlink/PoeChainlinkClient.sol";
+import "./FundingOracleClient.sol";
+import "@openzeppelin/contracts-ethereum-package/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts-ethereum-package/contracts/utils/Counters.sol";
 
-contract FundingFactory is Initializable {
+contract FundingFactory is Initializable,ERC721UpgradeSafe {
+    using Counters for Counters.Counter;
+    Counters.Counter private _tokenIds;
```

Comment:
I think if we want to support multiple ERC721 contracts if will be problematic to use our own Counter for them 🤔